pkg/cluster/cluster.go

```diff
@@ -5,13 +5,21 @@ package cluster
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"net/http"
+	"os"
+	"strings"
 	"time"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
```

Comment:
`go-autorest` is out of support: https://github.com/Azure/go-autorest

It looks like we're using `to` get string ptrs. Could we just write a super simple function to do this so we don't take a dependency on deprecated code?

Reply:
This is a nit comment

Reply:
I'll consider this out of scope for this PR since it would have to be changed all throughout the RP and not just here.

Reply:
It's not that I want you to fix this in this PR, but I think changing little by little is better than changing all at once because it will narrow the scope and we don't have to be bothered by conflicts.

Also https://pkg.go.dev/k8s.io/utils/ptr#To is already available in the codebase, so we can replace it now.

Reply:
I went ahead and switched to that function you pointed out since it's already available and easy enough 👍🏻